infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/brokerphases/DeployBroker.java

```diff
@@ -144,6 +154,28 @@ public DeployBroker(
     }
   }
 
+  private void watchLogsIfDebugEnabled(Map<String, String> startOptions, Pod pluginBrokerPod)
+      throws InfrastructureException {
+    if (startOptions == null) {
+      LOG.debug(
+          "'startOptions' is null so we won't watch the plugin broker pod logs for workspace '{}'",
+          runtimeId.getWorkspaceId());
+      return;
+    }
+    boolean shouldWatchContainerStartupLogs =
+        Boolean.parseBoolean(
+            startOptions.getOrDefault(DEBUG_WORKSPACE_START, Boolean.FALSE.toString()));
+
+    if (shouldWatchContainerStartupLogs) {
+      namespace
+          .deployments()
+          .watchLogs(
+              new PodLogHandlerToEventPublisher(runtimeEventsPublisher, runtimeId),
+              new LogWatchTimeouts(5_000, 100, 2_500),
```

Comment:
These should probably be extracted to constants or defaults somewhere.

Reply:
yes done